.pre-commit-config.yaml

```diff
@@ -68,6 +68,17 @@ repos:
         args: [--fix]
       # compatible replacement for black
       - id: ruff-format
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.9.0
+    hooks:
+      - id: mypy
+        language_version: "3.12"
+        args: [--follow-imports=silent, --strict]
+        additional_dependencies: [packaging >=23.0]
+        files: |
+          (?x)^(
+            conda/deprecations.py
+          )$
```

Comment:
Hmm, I think that should be discussed in the team first before we're adding it, not sure if everyone is familiar running mypy and how to resolve issues, also kind of not related to the PR

Reply:
the idea was that once a file was fully type annotated we wouldn't allow ourselves to regress

but yes I agree, its unrelated enough and should be extracted and put it into a new PR

Reply:
I agree with Ken that we should slowly enforce `mypy` over time.

Fair warning, I have had some inconsistencies with dependencies when running it through `pre-commit` over having `pre-commit` execute a `make` script that runs with the environment dependencies, but that's a story for another time.